packages/grafana-ui/src/components/Card/Card.tsx

```diff
@@ -243,3 +234,68 @@ export const getCardStyles = stylesFactory((theme: GrafanaTheme, disabled = fals
     `,
   };
 });
+
+interface ChildProps {
+  styles?: ReturnType<typeof getCardStyles>;
+  disabled?: boolean;
+}
+
+const Tags: FC = ({ children }) => {
+  return <>{children}</>;
+};
+Tags.displayName = 'Tags';
+
+const Figure: FC<ChildProps> = ({ children, styles }) => {
+  return <div className={styles?.media}>{children}</div>;
+};
+
+Figure.displayName = 'Figure';
+
+const Meta: FC<ChildProps> = ({ children, styles }) => {
+  // Join meta data elements by '|'
+  const meta = useMemo(
+    () =>
+      Array.isArray(children)
+        ? React.Children.toArray(children)
+            .filter(Boolean)
+            .reduce((prev, curr, i) => [
+              prev,
+              <span key={`separator_${i}`} className={styles?.separator}>
+                |
+              </span>,
+              curr,
+            ])
+        : children,
+    [children, styles?.separator]
+  );
+  return <div className={styles?.metadata}>{meta}</div>;
+};
+
+Meta.displayName = 'Meta';
+
+interface ActionsProps extends ChildProps {
+  children: JSX.Element[];
+}
+const Actions: FC<ActionsProps> = ({ children, styles, disabled }) => {
+  return (
+    <div className={styles?.actions}>{React.Children.map(children, child => cloneElement(child, { disabled }))}</div>
```

Comment:
why React.Children needed here? why do we need to pass this disabled property it does not look like it's a property that is calculated internally but passed from the outside

Reply:
We disable actions if the card has `disabled` prop and that needs to be passed to each action.

Reply:
but the actions are rendered outside so the user can set that. Guess it does not hurt (if the component is not a button that takes in this property)

Reply:
The idea is that the Card itself handles disabled logic. If there actions, they get disabled, otherwise the Card itself is disabled. Delegating part of this logic to the actions will makes things more complicated. Plus there are secondary actions that also get this prop.

Reply:
makes sense, just found it strange passing a prop from outside to the children when the consumer owns both elements & state & properties. A bit hidden logic that the Card component clones the element you gave it and sets this new property that the consumer did not set. But I can see that it's nice as well